massdriver.yaml

```diff
@@ -0,0 +1,69 @@
+schema: draft-07
+name: aws-sagemaker-inference-endpoint
+description: AWS SageMaker Inference Endpoint for hosting an AI model
+source_url: github.com/massdriver-cloud/aws-sagemaker-inference-endpoint
+access: public
+type: infrastructure
+
+params:
+  required:
+    - endpoint_config
+  properties:
+    endpoint_config:
+      title: Endpoint Configuration
+      type: object
+      required:
+        - primary_container
+        - instance_type
+        - instance_count
+      properties:
+        primary_container:
+          title: Primary Container
+          type: object
+          required:
+            - model_data
+            - ecr_image
+          properties:
+            model_data:
+              title: Model Data S3 URI
+              description: "The S3 URI of the model data. (e.g. s3://bucket-name/model.tar.gz)"
+              type: string
```

Comment:
You can use the `pattern` directive and specify a regex that the field must match. Here is a regex for a string that must start with `s3://` and must end with `.tar.gz`:

```
^s3:\/\/.+\.tar\.gz$
```

Then in the `message.pattern` field you can specify the error message that will be displayed if it violates the pattern.

Here is an example from GKE for a CIDR: https://github.com/massdriver-cloud/gcp-gke-cluster/blob/main/massdriver.yaml#L68-L70

Reply:
Added the regex for S3 and ECR